src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_PerfInitPage.js

```diff
@@ -395,37 +571,148 @@ class CJ4_FMC_PerfInitPage {
+            : crosswind < 0 ? "T"
+            : "";
+			headwind = Math.abs(headwind);
+			crosswind = Math.abs(crosswind);
+        }
+        let arrRunwayConditionActive = fmc.arrRunwayCondition == 0 ? "DRY"
+        : "WET";
         fmc.setTemplate([
-            ["APPROACH REF[color]blue", "1", "3"],
+            [destinationIdent + "   APPROACH REF[color]blue", "1", "3"],
 			["SEL APT[color]blue", "WIND[color]blue"],
-            ["depRunway", "---\xB0/---"],
+            [destinationIdent + "/" + originIdent + "[color]green", fmc.landingWindDir + "\xB0/" + fmc.landingWindSpeed],
             ["RWY ID[color]blue", "OAT[color]blue"],
-            ["---", "□□□\xB0C"],
+            [arrRunwayDesignation, fmc.landingOat + "\xB0C"],
             ["RWY WIND[color]blue", "QNH[color]blue"],
-            ["H4  L0", "placehold"],
+            [headwindDirection + headwind + " " + crosswindDirection + crosswind, fmc.landingQnh],
             ["RUNWAY LENGTH[color]blue", "P ALT[color]blue"],
-            ["5001 FT", "placehold"],
+            [Math.round(arrRunwayLength) + " FT", fmc.landingPressAlt + " FT"],
             ["RWY SLOPE[color]blue"],
             ["--.-%"],
             ["RWY COND[color]blue"],
-            ["DRY[color]green" + "/WET"]
+            [arrRunwayConditionActive + "[color]green"]
         ]);
+		fmc.onRightInput[0] = () => {
+            fmc.landingWindDir = fmc.inOut.slice(0, 3);
+			fmc.landingWindSpeed = fmc.inOut.slice(4, 7);
+			fmc.clearUserInput();
+			{ CJ4_FMC_PerfInitPage.ShowPage13(fmc); };
+		}
+		fmc.onRightInput[1] = () => {
+            fmc.landingOat = new Number(fmc.inOut);
+			fmc.clearUserInput();
+			{ CJ4_FMC_PerfInitPage.ShowPage13(fmc); };
+		}
+		fmc.onRightInput[2] = () => {
+			fmc.landingQnh = new Number(fmc.inOut).toFixed(2);
+			fmc.landingPressAlt = Math.trunc((((29.92 - fmc.landingQnh) * 1000) + arrRunwayElevation)) + " FT";
+			fmc.clearUserInput();
+			{ CJ4_FMC_PerfInitPage.ShowPage13(fmc); };
+		}
+        fmc.onLeftInput[5] = () => {
+            if (fmc.arrRunwayCondition == 0) {
+                fmc.arrRunwayCondition = 1;
+            } else if (fmc.arrRunwayCondition == 1) {
+                fmc.arrRunwayCondition = 0;
+            }
+            arrRunwayConditionActive = fmc.arrRunwayCondition == 0 ? "DRY"
+                : "WET";
+				fmc.clearUserInput();
+			{ CJ4_FMC_PerfInitPage.ShowPage13(fmc); };
+        }
 		fmc.onPrevPage = () => { CJ4_FMC_PerfInitPage.ShowPage15(fmc); };
         fmc.onNextPage = () => { CJ4_FMC_PerfInitPage.ShowPage14(fmc); };
         fmc.updateSideButtonActiveStatus();
     }
 	static ShowPage14(fmc) { //APPROACH REF Page 2
         fmc.clearDisplay();
+		let grWtCell = "";
+        let grossWeightValue = fmc.getWeight();
+        if (isFinite(grossWeightValue)) {
+            grWtCell = (grossWeightValue * 2200).toFixed(0);
+        }
+		let vRef = ((grWtCell - 10500) * .00393) + 92; //V Speeds based on weight at 0C
```

Comment:
Is this the same vSpeeds code as elsewhere?

Reply:
I think the v speed code for takeoff and landing are different math and values and in their own pages. Each are only used on the one page each, so I think they are good to leave in the individual page definitions.